diesel/src/mysql/connection/bind.rs

```diff
@@ -213,22 +218,63 @@ impl BindData {
-    use self::ffi::enum_field_types::*;
+impl From<MysqlType> for ffi::enum_field_types {
+    fn from(tpe: MysqlType) -> Self {
+        use self::ffi::enum_field_types::*;
+
+        match tpe {
+            MysqlType::Tiny => MYSQL_TYPE_TINY,
+            MysqlType::Short => MYSQL_TYPE_SHORT,
+            MysqlType::Long => MYSQL_TYPE_LONG,
+            MysqlType::LongLong => MYSQL_TYPE_LONGLONG,
+            MysqlType::Float => MYSQL_TYPE_FLOAT,
+            MysqlType::Double => MYSQL_TYPE_DOUBLE,
+            MysqlType::Time => MYSQL_TYPE_TIME,
+            MysqlType::Date => MYSQL_TYPE_DATE,
+            MysqlType::DateTime => MYSQL_TYPE_DATETIME,
+            MysqlType::Timestamp => MYSQL_TYPE_TIMESTAMP,
+            MysqlType::String => MYSQL_TYPE_STRING,
+            MysqlType::Blob => MYSQL_TYPE_BLOB,
+            MysqlType::Numeric => MYSQL_TYPE_NEWDECIMAL,
+        }
+    }
+}
 
-    match tpe {
-        MysqlType::Tiny => MYSQL_TYPE_TINY,
-        MysqlType::Short => MYSQL_TYPE_SHORT,
-        MysqlType::Long => MYSQL_TYPE_LONG,
-        MysqlType::LongLong => MYSQL_TYPE_LONGLONG,
-        MysqlType::Float => MYSQL_TYPE_FLOAT,
-        MysqlType::Double => MYSQL_TYPE_DOUBLE,
-        MysqlType::Time => MYSQL_TYPE_TIME,
-        MysqlType::Date => MYSQL_TYPE_DATE,
-        MysqlType::DateTime => MYSQL_TYPE_DATETIME,
-        MysqlType::Timestamp => MYSQL_TYPE_TIMESTAMP,
-        MysqlType::String => MYSQL_TYPE_STRING,
-        MysqlType::Blob => MYSQL_TYPE_BLOB,
+impl From<ffi::enum_field_types> for MysqlType {
+    fn from(tpe: ffi::enum_field_types) -> Self {
+        use self::ffi::enum_field_types::*;
+
+        match tpe {
+            MYSQL_TYPE_TINY => MysqlType::Tiny,
+            MYSQL_TYPE_SHORT => MysqlType::Short,
+            MYSQL_TYPE_INT24 | MYSQL_TYPE_LONG => MysqlType::Long,
+            MYSQL_TYPE_LONGLONG => MysqlType::LongLong,
+            MYSQL_TYPE_FLOAT => MysqlType::Float,
+            MYSQL_TYPE_DOUBLE => MysqlType::Double,
+            MYSQL_TYPE_TIME => MysqlType::Time,
+            MYSQL_TYPE_DATE => MysqlType::Date,
+            MYSQL_TYPE_DATETIME => MysqlType::DateTime,
+            MYSQL_TYPE_TIMESTAMP => MysqlType::Timestamp,
+            MYSQL_TYPE_STRING => MysqlType::String,
+            MYSQL_TYPE_BLOB => MysqlType::Blob,
+            MYSQL_TYPE_DECIMAL | MYSQL_TYPE_NEWDECIMAL => MysqlType::Numeric,
+            MYSQL_TYPE_NULL
+            | MYSQL_TYPE_YEAR
+            | MYSQL_TYPE_NEWDATE
+            | MYSQL_TYPE_VARCHAR
+            | MYSQL_TYPE_BIT
+            | MYSQL_TYPE_TIMESTAMP2
+            | MYSQL_TYPE_DATETIME2
+            | MYSQL_TYPE_TIME2
+            | MYSQL_TYPE_JSON
+            | MYSQL_TYPE_ENUM
+            | MYSQL_TYPE_SET
+            | MYSQL_TYPE_TINY_BLOB
+            | MYSQL_TYPE_MEDIUM_BLOB
+            | MYSQL_TYPE_LONG_BLOB
+            | MYSQL_TYPE_VAR_STRING
+            | MYSQL_TYPE_GEOMETRY => unimplemented!(),
```

Comment:
The question is how do we want to handle this case? Implementing `TryFrom` would imply that we change the return types in `Row` and `NamedRow`

Reply:
Honestly I think we need some actual test cases demonstrating that we even can get those types in the first place to really answer this. Everything ending with `2` we should never get. I suspect everything that just has some modifier like `LONG` or `VAR` we can treat identically to the alternatives. Not sure on types like geometry and json. Need to dig more to comment on your statement about From vs TryFrom

Reply:
https://dev.mysql.com/doc/refman/8.0/en/json.html and https://dev.mysql.com/doc/refman/8.0/en/spatial-types.html could be relevant here for json and geometry types

Reply:
https://docs.oracle.com/cd/E17952_01/mysql-8.0-en/c-api-data-structures.html has at least some information about `MYSQL_TYPE_GEOMETRY` (and some of the other unhandled types)